execution_engine/src/shared/chain_kind.rs

```diff
@@ -0,0 +1,72 @@
+//! Definition of a chain kind flag that is used to determine a operating mode of a chain.
+use datasize::DataSize;
+use num_derive::FromPrimitive;
+use num_traits::FromPrimitive;
+use rand::{distributions::Standard, prelude::*};
+use serde::{Deserialize, Serialize};
+
+use casper_types::bytesrepr::{Error, FromBytes, ToBytes, U8_SERIALIZED_LENGTH};
+
+/// Flag representing a mode of operation of a chain.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize, DataSize, FromPrimitive)]
+#[serde(rename_all = "lowercase")]
+pub enum ChainKind {
+    /// Public (default) mode
+    Public,
+    /// Private chain mode with extra capabilities only available if this mode is set.
+    Private,
```

Comment:
Each private chain could choose a different set of "special features". I don't think we should generalize and assume that all of them will want the same thing.